pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+const supportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	newVersionMajor := newVersion.Major()
+	newVersionMinor := newVersion.Minor()
+	oldVersionMajor := oldVersion.Major()
+	oldVersionMinor := oldVersion.Minor()
+
+	majorVersionDifference := math.Abs(float64(newVersionMajor) - float64(oldVersionMajor))
+
+	if majorVersionDifference > 0 {
+		return fmt.Errorf("MAJOR version upgrades are not supported. Major version difference between upgrade version (%d.%d) and server version (%d.%d) is %f", newVersionMajor, newVersionMinor, oldVersionMajor, oldVersionMinor, majorVersionDifference)
```

Comment:
The fact that we need to do `(%d.%d)` makes me thing that this method should probably take `anywherev1.KubernetesVersion` instead of `*version.Version`

Reply:
To make this method common, I am sending in the `*version.Version`. For the CLI we get the server version from the cluster which is usually of format similar to `v1.18.16-eks-1-18-4`. With the webhook however both formats are of `1.18`. We use `ParseGeneric` for webhook vs `ParseSemantic` and `ParseGeneric` for the CLI.